Fix Chat stack of images opening in focus mode (#979)

NikeshNayak opened the pull request:
Resolves #979   

## Synopsis

When a message with an image is focused (with a long tap on mobile). Whenever we tap on the image, it is opening a image in background in the focus mode, so multiple taps create a stack of images. Fullscreen Gallery images are opening in the background.

## Solution

We have disable the tap behaviour on image when message is focused, so it stops images getting open in the background.

## Checklist

- Created PR:
    - [X] In [draft mode][l:1]
    - [X] Name contains issue reference
    - [X] Has type and `k::` labels applied
- Before [review][l:4]:
    - [X] Documentation is updated (if required)
    - [X] Tests are updated (if required)
    - [X] Changes conform [code style][l:2]
    - [x] [CHANGELOG entry][l:3] is added (if required)
    - [x] FCM (final commit message) is posted or updated
    - [x] [Draft mode][l:1] is removed
- [x] [Review][l:4] is completed and changes are approved
    - [x] FCM (final commit message) is approved
- Before merge:
    - [x] Milestone is set
    - [x] PR's name and description are correct and up-to-date
    - [x] All temporary labels are removed
    


[l:1]: https://help.github.com/en/articles/about-pull-requests#draft-pull-requests
[l:2]: /CONTRIBUTING.md#code-style
[l:3]: /CHANGELOG.md
[l:4]: https://help.github.com/en/articles/reviewing-changes-in-pull-requests

NikeshNayak commented:
FCM

```
Disable tap behaviour on image when message is focused (#983, #979)  
```

SleepySquash commented:
@NikeshNayak, why don't you merge the latest `main` branch into yours before sending it for a review? It's not a problem, when no conflicts are present, yet when there are, you should be the one most familiar with the code to resolve it ideally, don't you agree?

Since you've already sent it for a review, you shouldn't do any commits now, yet please bear that in mind the future PRs.

SleepySquash commented on lib/ui/page/home/page/chat/widget/chat_item.dart:
```diff
@@ -245,6 +246,7 @@ class ChatItemWidget extends StatefulWidget {
           onTap: isLocal
               ? null
               : () {
+                  if (disableGallery) return;
```
This doesn't disable the tap itself. The tap is possible, it's just its handler returns immediately. Don't you think disabling the gesture would be a cleaner approach?

NikeshNayak replied:
@SleepySquash Yes, but disabling tap behaviour on gesture was closing the context menu when I tap on the image. Will it be okay to do that ?

SleepySquash replied:
@NikeshNayak, ah, I see, that's ok then. You should add some comments describing code, when something not really trivial happens. I'll continue the review then.

NikeshNayak commented:
> @NikeshNayak, why don't you merge the latest `main` branch into yours before sending it for a review? It's not a problem, when no conflicts are present, yet when there are, you should be the one most familiar with the code to resolve it ideally, don't you agree?
> 
> Since you've already sent it for a review, you shouldn't do any commits now, yet please bear that in mind the future PRs.

I apologize for that, I will keep in mind for the future PRs.

SleepySquash commented:
@NikeshNayak, I can see the links in the description of PR being broken. Why is that?

> Before [review][l:4]:
> [CHANGELOG entry][l:3]
> [Draft mode][l:1] is removed

SleepySquash commented on CHANGELOG.md:
```diff
@@ -16,9 +16,13 @@ All user visible changes to this project will be documented in this file. This p
 - UI:
     - Home page:
         - Own avatar displaying with question marks when signing in or signing up. ([#978], [#950])
+    - Chat page:
+        - Tapping on image opens gallery when being in context menu. ([#983], [#979])
```
>  [ ] [CHANGELOG entry][l:3] is added (if required)

Why don't you do it? It's your responsibility to add changelog entries for end-user visible changes.

SleepySquash commented on lib/ui/page/home/page/chat/widget/chat_forward.dart:
```diff
@@ -440,7 +440,7 @@ class _ChatForwardWidgetState extends State<ChatForwardWidget> {
                     galleries,
                     key: _galleryKeys[msg.id]?.firstOrNull,
-                    onGallery: widget.onGallery,
+                    onGallery: menu ? null : widget.onGallery,
```
You forgot about `ChatForward`s - those types of `ChatItem`s are displayed via separate widget due to complexity of forwards.

NikeshNayak commented:
> @NikeshNayak, I can see the links in the description of PR being broken. Why is that?
> 
> > Before [review][l:4]:
> > [CHANGELOG entry][l:3]
> > [Draft mode][l:1] is removed

@SleepySquash Yes, Unfortunately I removed it while editing the description. I have fixed now.

SleepySquash commented on lib/ui/page/home/page/chat/widget/chat_item.dart:
```diff
@@ -245,8 +246,13 @@ class ChatItemWidget extends StatefulWidget {
               ? null
               : () {
-                  final Iterable<GalleryAttachment> attachments =
-                      onGallery?.call() ?? media;
+                  if (onGallery == null) {
+                    // [onTap] still needs to be invoked to ensure [ContextMenu]
+                    // doesn't get closed, when this is being built within it.
+                    return;
+                  }
```
I think `null`ifying the `onGallery` callback is a bit cleaner than adding separate flag for preventing the `GalleryPopup` from being opened, what's yours thoughts?

NikeshNayak replied:
@SleepySquash Yes, I think that would be better approach than `disableGallery` flag
